kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -785,7 +800,21 @@ class TableFeaturesSuite extends AnyFunSuite {
         7,
         set("columnMapping", "deletionVectors"),
         set("columnMapping", "icebergCompatV2", "deletionVectors")),
-      set("icebergCompatV2"))).foreach {
+      set("icebergCompatV2")),
+    (
+      testMetadata(tblProps = Map("delta.enableIcebergWriterCompatV1" -> "true")),
+      new Protocol(1, 2),
+      new Protocol(
+        2,
+        7,
+        set(),
+        set(
+          "columnMapping",
+          "appendOnly",
+          "invariants",
+          "icebergCompatV2",
+          "icebergWriterCompatV1")),
+      set("icebergCompatV2", "columnMapping", "icebergWriterCompatV1"))).foreach {
```

Comment:
one more test case when input protocol is (2, 7) and (3, 7) (reader/writer is pre-post feature support)?